File: src/hausaufgabe.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main()
{
    FILE* ptr;
    char ch;

    ptr = fopen("secret_data.txt", "r");

    if (NULL == ptr) {
        printf("Die Datei konnte nicht geöffnet werden.\n");
        return 1;
    }

    printf("Login mit Daten:\n")

    do {
        ch = fgetc(ptr);
        if (ch != EOF) {
            putchar(ch);
        }

    } while (ch != EOF);

    fclose(ptr);
    return 0;
}

File: src/hausaufgabe.py
def main() -> None:
    try:
        with open("secret_data.txt", "w") as file:
            print("Login mit Daten:\n")
            while True:
                ch = file.read(1)
                if not ch:
                    break
                print(ch, end="")
    except ValueError:
        print("Die Datei konnte nicht geöffnet werden.")
        return 1
    return 0

if __name__ == "__main__":
    print("Nichts passiert. :( Ich glaube, du musst noch einige Fehler beheben.")
